src/encoded/static/components/motifs.js

```diff
@@ -285,6 +352,60 @@ export const Motifs = (props) => {
         alignedEndCoordinate = Math.max(p.end, alignedEndCoordinate);
     });
 
+    // trimming the reference sequence to widest window of pwms
+    const referenceLength = alignedEndCoordinate - alignedStartCoordinate;
+    const referenceStart = alignedStartCoordinate - props.context.sequence.start;
+    const referenceSequence = props.context.sequence.sequence.slice(referenceStart, referenceStart + referenceLength);
+
+    const trackScrolling = () => {
+        if (tableTop && tableTop.current && refContainer && refContainer.current && refPlaceholder && refPlaceholder.current) {
+            if (tableTop.current.getBoundingClientRect().top <= 27) {
+                refContainer.current.classList.add('fixed-position');
+                refPlaceholder.current.setAttribute('style', `height:${refContainer.current.getBoundingClientRect().height}px`);
+            } else {
+                refContainer.current.classList.remove('fixed-position');
+                refPlaceholder.current.setAttribute('style', 'width:0px');
+            }
+        }
+    };
+
+    React.useEffect(() => {
+        if (limit === 0) {
+            document.addEventListener('scroll', trackScrolling);
+        }
+
+        // drawing reference sequence as a pwm so it lines up with the others
+        require.ensure(['d3', 'd3-sequence-logo'], (require) => {
+            setD3Lib(require('d3'));
+            const sequenceLogos = logos; // This is for local development when changes are needed to d3-sequence-logo.
+            // this.sequenceLogos = require('d3-sequence-logo');
+
+            const refSeq = referenceSequence.split('');
+            const fakePWM = [];
+
+            refSeq.forEach((nucleotide) => {
+                const maxHeight = 1000;
+                if (nucleotide === 'A') {
+                    fakePWM.push([maxHeight, 0, 0, 0]);
+                } else if (nucleotide === 'C') {
+                    fakePWM.push([0, maxHeight, 0, 0]);
+                } else if (nucleotide === 'G') {
+                    fakePWM.push([0, 0, maxHeight, 0]);
+                } else {
+                    fakePWM.push([0, 0, 0, maxHeight]);
+                }
+            });
```

Comment:
I bet you can use a `.map()` here which might look more clear, so `const fakePWM = refSeq.map((nucleotide) => {...`. But `map`s an alternative, not a correction. I try to use it instead of `forEach` when possible, and it looks like `fakePWM` and `refSeq` have a one-to-one mapping, so I think it’s possible here.